substrings.rb

```diff
@@ -0,0 +1,8 @@
+def substrings(string, dictionary)
+    count_hash = {}
+    dictionary.map {|substring| count_hash[substring] = string.downcase.scan(/#{substring.downcase}/).count }
+    count_hash
```

Comment:
I think with the check added as requested [here](https://github.com/yaeluria/ruby-assignments/pull/3#discussion_r893422393) in number 5, it's best to use `each_with_object ` so I can set a variable within the loop, and reduce is not as good because you still need to call the object to return it.